website/blog/2024-06-25-use-a-framework-to-build-react-native-apps.md

```diff
@@ -0,0 +1,68 @@
+
+<!-- truncate  -->
+
+## What is a React Native framework?
+
+If you’ve been building production-ready apps, you probably know that there is a set of common problems you will need to solve sooner or later.
+
+When building any application on either web or native, you probably want your users to navigate through different screens, fetch data, and store the state of your user. But for native apps there is even more to deal with: you need tools to upgrade your native code between React Native versions, manage compatible versions of all of your dependencies, and deal with native build tools.
+
+It takes a village to bring an app from idea to production without the right tools.
+
+We want you to focus on writing beautiful applications and features for your users, and not solving those common problems over and over.
+
+That’s why we believe that the best way for you to experience React Native is through a framework that offers a toolbox with all the necessary tools you need to build production-ready applications.
+
+We’ve found that **you’re either using a framework or you’re building your own framework**.
+
+There is nothing wrong with building your own framework, by crafting your own solutions for routing, navigation, deploying, and so on. Major corporations like Meta and Microsoft build their own frameworks internally to integrate deeply into their brownfield apps. But we believe that most people will be better off by using an existing framework.
+
+If you’ve been using React on web, you’re probably familiar with a similar concept of [production-grade React frameworks](https://react.dev/learn/start-a-new-react-project#production-grade-react-frameworks).
+
+As of today, the only recommended community framework for React Native is [Expo](https://docs.expo.dev/). Folks at Expo have been investing in the React Native ecosystem since the early days of React Native and as of today, we believe the developer experience offered by Expo is best in class.
+
+:::note
+
+Expo, the framework, is and will remain free and open source, while Expo Application Services (EAS) is an optional paid service.
+
+:::
+
+<!--alex ignore he-she retext-equality-->
+
+If you haven’t used Expo recently, make sure you don’t miss [this talk from Kadi @ Expo](https://www.youtube.com/live/0ckOUBiuxVY?si=N-WSfmAJSMfd6wDL&t=3888) where she’s showcasing what you can do with Expo in 2024.
+
+We’ve also updated the [Getting Started page](https://reactnative.dev/docs/environment-setup) on the website to reflect this recommendation.
+
+## How will frameworks affect you?
+
+If you’re already using a recommended framework such as Expo, you’re already good to go!
+
+If you’re not using Expo, the impact will be minimal. If you like, you can consider migrating, and you can find instructions on[ the official Expo website](https://docs.expo.dev/bare/overview/).
```

Comment:
This part is worth expanding a bit IMHO. "the impact will be minimal" does not really give any clue of the impact, nor if it's positive or negative. Maybe worth mentioning the migration can be incremental, and list a few concrete immediate benefits you get when adopting Expo?

Reply:
"the impact will be minimal" -> Practically nothing will change as we implemented everything in a backward compatible way. The only thing that will change is that we'll be deprecating the `react-native init` command in favor of either CLI init or expo-init.

Reply:
> does not really give any clue of the impact, nor if it's positive or negative

It's listed below in:

> The tools you’ve been using such as React Native Community CLI, Template and [Upgrade Helper](https://react-native-community.github.io/upgrade-helper/) will keep on working as usual.

Reply:
I added a suggestion for how this could be made more concrete, by articulating that the CLI has moved. Consider if there are other things you'd want to say explicitly